client/authority-discovery/src/error.rs

```diff
@@ -46,4 +46,8 @@ pub enum Error {
 	EncodingDecodingScale(codec::Error),
 	/// Failed to parse a libp2p multi address.
 	ParsingMultiaddress(libp2p::core::multiaddr::Error),
+	/// Failed to sign using a specific public key
+	MissingSignature(usize),
+	/// Failed to sign using all public keys
+	SigningFailed,
```

Comment:
None of the error names above are postfixed with `Failed`. I don't care about the name but would prefer to stay consistent, e.g.:
```suggestion
	Signing,
```